### PR TITLE
Feature/autosuggest material

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue
image
image

#### Description
Autosuggest-material is a subcomponent of the autosuggest component used to display material suggestions in an autosuggest dropdown. Here is a link to the Figma design file for [desktop](https://www.figma.com/file/ETOZIfmgGS1HUfio57SOh7/S%C3%B8gning?node-id=4709%3A24976), and [mobile](https://www.figma.com/file/ETOZIfmgGS1HUfio57SOh7/S%C3%B8gning?node-id=1749%3A17989) version of the component.

#### Screenshot of the result
Desktop: 
image

Mobile: 
image

#### Additional comments or questions
What do you think of the solution and of the implementation?

### CODE REVIEW

File: src/stories/autosuggest-material/AutosuggestMaterial.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    year: string;
+  }[]
+}
+
+export const AutosuggestMaterial = (props: AutosuggestMaterialProps) => {
+  const {items} = props
+
+  return (
+    <ul className="autosuggest">
+      <li className="autosuggest__item--text text-body-medium-regular">
```
Comment (spaceo) [inline src/stories/autosuggest-material/AutosuggestMaterial.tsx:17]: Do something

File: src/stories/autosuggest-text/AutosuggestText.tsx [hunk 2]
```diff
@@ -5,10 +5,10 @@ export type AutosuggestTextProps = {
 export const AutosuggestText = (props: AutosuggestTextProps) => {
   const { items } = props;
   return (
-    <ul className="autosuggest-text-container">
+    <ul className="autosuggest">
       {items.map((item) => {
         return (
-          <li className="autosuggest-text-container__item text-body-medium-regular">
+          <li className="autosuggest__item--text text-body-medium-regular">
```
Comment (spaceo) [inline src/stories/autosuggest-text/AutosuggestText.tsx:11]: Do something here as well?

File: src/stories/autosuggest-text/autosuggest-text.scss [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
+.autosuggest {
   background-color: $c-global-primary;
   outline: 1px solid $c-global-tertiary-1;
   position: absolute;
   left: 0;
   right: 0;
-  top: calc(100% + 2px);
-  padding-bottom: 30px;
+  top: calc(100% + 1px);
+  padding-bottom: 10px;
```
Comment (spaceo) [inline src/stories/autosuggest-text/autosuggest-text.scss:8]: We dont have fixed spacing variables?

File: src/stories/autosuggest-text/autosuggest-text.scss [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
-    padding: 0 23px;
 
-    &:hover {
-      background-color: $c-global-secondary;
-      cursor: pointer;
+    &--text {
+      height: 40px;
+      display: flex;
+      align-items: center;
+      padding: 0 23px;
```
Comment (spaceo) [inline src/stories/autosuggest-text/autosuggest-text.scss:20]: Dont we have fixed spacing variables?

**Review by spaceo — CHANGES_REQUESTED** [review]
Look great! I have a few comments ;)

File: src/stories/autosuggest-text/autosuggest-text.scss (hunk 4 above)
Comment (Adamik10) [reply src/stories/autosuggest-text/autosuggest-text.scss:20]: yes! :) gotcha

File: src/stories/autosuggest-material/AutosuggestMaterial.tsx (hunk 1 above)
Comment (Adamik10) [reply src/stories/autosuggest-material/AutosuggestMaterial.tsx:17]: roger that! :)

File: src/stories/autosuggest-text/AutosuggestText.tsx (hunk 2 above)
Comment (Adamik10) [reply src/stories/autosuggest-text/AutosuggestText.tsx:11]: yes yes :)

**Review by spaceo — APPROVED** [review]

File: src/stories/autosuggest-material/AutosuggestMaterial.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+export default {
+  title: "Components / Autosuggest - Material",
+  component: AutosuggestMaterial,
+  decorators: [withDesign],
+  argTypes: {
+    items: {
+      name: "Items",
+      defaultValue: [
+        {
+          cover: "https://imgcdn.saxo.com/_9780345514400/0x500",
```
[the remark this answers is not in the document]
Comment (Adamik10) [reply src/stories/autosuggest-material/AutosuggestMaterial.stories.tsx:14]: Oh yes, for sure :) I just felt they didn't really inspire me.

File: src/stories/autosuggest-material/AutosuggestMaterial.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            I am a contextual text item
+      </li>
+      <hr className="autosuggest__divider"/>
+      <ul className="autosuggest__materials">
```
[the remark this answers is not in the document]
Comment (Adamik10) [reply src/stories/autosuggest-material/AutosuggestMaterial.tsx]: That's very correct! Thanks ❤️ I fixed it

File: src/stories/autosuggest-material/AutosuggestMaterial.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        {items.map((item) => {
+          return (
+            <li className="autosuggest__materials__item">
+              <div className="autosuggest__materials__item__content">
+                <div className="autosuggest__materials__item__content__cover">
```
[the remark this answers is not in the document]
Comment (Adamik10) [reply src/stories/autosuggest-material/AutosuggestMaterial.tsx]: Haha yes! I followed this article to fix it: https://scalablecss.com/bem-nesting-grandchild-elements/

**Review by sinejespersen — APPROVED** [review]

**Review by cableman — CHANGES_REQUESTED** [review]
I don't think we should add covers to the repositories (also we don't have the copyright to the covers?)

**Review by cableman — APPROVED** [review]

**Pull request merged**